framework/src/modules/chain/components/storage/entities/chain_meta.js

```diff
@@ -0,0 +1,183 @@
+
+/**
+ * ChainMeta
+ * @typedef {Object} ChainMeta
+ * @property {string} key
+ * @property {string} value
+ */
+
+/**
+ * ChainMeta Filters
+ * @typedef {Object} filters.ChainMeta
+ * @property {string} [key]
+ * @property {string} [key_eql]
+ * @property {string} [key_ne]
+ */
+
+class ChainMeta extends BaseEntity {
+	constructor(adapter, defaultFilters = {}) {
+		super(adapter, defaultFilters);
+
+		this.addField('key', 'string', { filter: BOOLEAN });
+		this.addField('value', 'string');
+
+		this.sqlDirectory = path.join(path.dirname(__filename), '../sql');
+		this.SQLs = this.loadSQLFiles('chain_meta', sqlFiles, this.sqlDirectory);
+	}
+
+	/**
+	 * Get list of meta information
+	 *
+	 * @param {filters.ChainMeta|filters.ChainMeta[]} [filters = {}]
+	 * @param {Object} [options = {}] - Options to filter data
+	 * @param {Number} [options.limit=10] - Number of records to fetch
+	 * @param {Number} [options.offset=0] - Offset to start the records
+	 * @param {Object} [tx] - Database transaction object
+	 * @return {Promise.<ChainMeta[], Error>}
+	 */
+	get(filters = {}, options = {}, tx = null) {
+		return this._getResults(filters, options, tx);
+	}
+
+	/**
+	 * Get list of meta information
+	 *
+	 * @param {filters.ChainMeta|filters.ChainMeta[]} [filters = {}]
+	 * @param {Object} [options = {}] - Options to filter data
+	 * @param {Number} [options.limit=10] - Number of records to fetch
+	 * @param {Number} [options.offset=0] - Offset to start the records
+	 * @param {Object} [tx] - Database transaction object
+	 * @return {Promise.<ChainMeta, Error>}
+	 */
+	getOne(filters = {}, options = {}, tx = null) {
+		const expectedResultCount = 1;
+		return this._getResults(filters, options, tx, expectedResultCount);
+	}
+
+	/**
+	 * Fetch the key value or resolve to null
+	 *
+	 * @param {string} key - Key to fetch
+	 * @param {Object} [tx] - Database transaction object
+	 * @return {Promise<{key: string, value: string} | {}>}
+	 */
+	async getKey(key, tx) {
+		assert(key, 'Must provide the key to get');
+
+		return this.getOne({ key }, {}, tx)
+			.then(data => data.value)
+			.catch(error => {
+				if (error.message === 'No data returned from the query.') {
+					return null;
+				}
+
+				throw error;
+			});
+	}
+
+	/**
+	 * Update or create the key value pair
+	 *
+	 * @param {string} key
+	 * @param {string} value
+	 * @param {Object} [tx] - Database transaction object
+	 * @return {Promise.<null, Error>}
+	 */
+	async setKey(key, value, tx) {
+		assert(key, 'Must provide the key to set');
+		assert(value, 'Must provide the value to set');
+
+		const expectedResultCount = 0;
+
+		return this.adapter.executeFile(
+			this.SQLs.upsert,
+			{ key, value },
+			{ expectedResultCount },
+			tx
+		);
+	}
+
+	/**
+	 * Delete the keys with following conditions
+	 *
+	 * @param {filters.ChainMeta} filters
+	 * @param {Object} [options]
+	 * @param {Object} [tx]
+	 * @returns {Promise.<boolean, Error>}
+	 */
+	delete(filters, _options, tx = null) {
```

Comment:
no `deleteKey`?

Reply:
`get`, `getOne`, `delete` are consistent interfaces from BaseEntity, so used the same.